tests/test_EclipseIO.cpp

```diff
@@ -431,3 +435,309 @@ WELSPECS
+         }
+
+         deckString += std::string{ R"(/
+)"};
+     }
+     std::vector<double> edit_mult_x(18, 1);
+     std::vector<double> edit_mult_z(18, 1);
+     if (doxyz[0].test(0) || doxyz[2].test(1))
+     {
+         // add an EDIT section
+         deckString += std::string{ R"(
+EDIT
+)"};
+
+         if (doxyz[0].test(0))
+         {
+             deckString += std::string{ R"(
+MULTX
+ 18*100 /
+MULTX
+ 18*0.1 /
+)"};
+             edit_mult_x.assign(18, .1);
+         }
+
+         if (doxyz[2].test(1))
+         {
+             deckString += std::string{ R"(
+MULTZ-
+ 18*30 /
+MULTZ-
+ 18*0.3 /
+)"};
+             edit_mult_z.assign(18, .3);
+         }
+         deckString += std::string{ R"(
+EQUALS
+)"};
+         if (doxyz[0].test(0))
+         {
+             deckString += std::string{ R"(
+ 'MULTX' 1.5 3 3 2 2 1 1 / -- Should not overwrite the value of GRID section but of EDIT section
+)"};
+             edit_mult_x[5] = 1.5;
+         }
+
+         if (doxyz[2].test(1))
+         {
+             deckString += std::string{ R"(
+ 'MULTZ-' 3.0 1 1  2 2 1 1 /
+)"};
+             edit_mult_z[3] = 3.0;
+         }
+
+             deckString += std::string{ R"(/
+)"};
+         auto mult = edit_mult_x.begin();
+         if (doxyz[0].test(0)) {
+             for (auto&& val: exspected_multipliers[0][0]) {
+                 val *= *(mult++);
+             }
+         }
+         mult = edit_mult_z.begin();
+         if (doxyz[2].test(1)) {
+             for (auto&& val: exspected_multipliers[2][1]) {
+                 val *= *(mult++);
+             }
+         }
+     }
+
+     return {deckString, exspected_multipliers};
+}
+
+/// \brief Test the MULTXYZ writing
+///
+/// \param doxyz Inidicates for each direction which multipliers should be test
+///              no bit set means none, 1st bit set positive, 2nd bit set negative
+/// \param write_all_mult_minus If true we will request that even defaulted MULT?- arrays will be
+///                  written via first record of GRIDOPTS
+void testMultxyz(std::array<std::bitset<2>,3> doxyz, bool write_all_multminus = false)
+{
+    const auto [deckString, exspectedMult] = createMULTXYZDECK(doxyz, write_all_multminus);
+    const auto deck = Parser().parseString(deckString);
+    auto es = EclipseState( deck );
+    const auto& eclGrid = es.getInputGrid();
+    const Schedule schedule(deck, es, std::make_shared<Python>());
+    const SummaryConfig summary_config( deck, schedule, es.fieldProps(), es.aquifer());
+    const SummaryState st(TimeService::now());
+    es.getIOConfig().setBaseName( "MULTXFOO" );
+    EclipseIO eclWriter( es, eclGrid , schedule, summary_config);
+    eclWriter.writeInitial( );
+
+    EclIO::EclFile initFile { "MULTXFOO.INIT" };
+
+    std::array<std::string, 6> multipliers{"MULTX", "MULTX-", "MULTY", "MULTY-", "MULTZ", "MULTZ-"};
+    int i=0;
+    auto mult = multipliers[i];
+    for (const auto& mult: multipliers) {
```

Comment:
The inner `mult` object shadows the outer here.  I suppose we could just remove the outer?

Reply:
Done